### PR TITLE
fix dasymetric and reorganize/cleanup

### PR DESCRIPTION
**Pull request by knaaptime**
resolves the bug uncovered in #165 using a different approach to carry out dasymetric masking (thanks @johnziebro!). 

Also:

- reogranizes the library a bit
- trims unused/old functions from `util` and elsewhere
- removes `glm_pixel_adjusted`, which was experimental to begin with and is causing testing headaches. If we want to return to this, there's probably a better (and faster) way using our new tooling

### CODE REVIEW

File: tobler/dasymetric/masked_area_interpolate.py [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
 
 
 def masked_area_interpolate(
     source_df,
     target_df,
-    raster="nlcd_2011",
-    codes=None,
-    force_crs_match=True,
+    raster,
+    pixel_values=[21, 22, 23, 24],
```
Comment (martinfleis) [inline tobler/dasymetric/masked_area_interpolate.py:11]: Shall we actually have a default here? This is specific NLCD data. While I understand that it was the dataset you often use with this function, it is by no means the expected dataset.

Comment (martinfleis) [reply tobler/dasymetric/masked_area_interpolate.py:11]: Also, this is a breaking change if we change `codes` to `pixel_values` that should have some deprecation period.

File: tobler/dasymetric/raster_tools.py [hunk 2]
```diff
@@ -27,6 +31,45 @@ def _apply_parser(df):
...
+    The generated geodataframe will input the value 0 for each Type that is not present in the raster
+    for each polygon.
+    """
+
+    _check_presence_of_crs(geodataframe)
+    if force_crs_match:
+        with rio.open(raster_path) as raster:
+            # raster =
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
```
Comment (martinfleis) [inline tobler/dasymetric/raster_tools.py:57]: Why do we catch warnings here? If there's a specific waring we want to suppress, we should catch that one. Not all.

Comment (knaaptime) [reply tobler/dasymetric/raster_tools.py]: i'm not sure why that's there. 

Agree, removed

File: tobler/dasymetric/masked_area_interpolate.py (hunk 1 above)
Comment (knaaptime) [reply tobler/dasymetric/masked_area_interpolate.py]: +2

**Review by martinfleis — APPROVED** [review]
Thanks!

**Pull request merged**